### ISSUE TITLE
exclusion non existing fields, properties

### ISSUE THREAD

**Issue by slawekjaranowski** [assertj/assertj-core#2217]
#### Summary

We can exclude fields, properties from some of checking like in `ObjectAssert#hasNoNullFieldsOrPropertiesExcept`, `RecursiveComparisonAssert#ignoringFields` (maybe in other place also)

We can misspell in field name or filed can be removed from object in future development, so assertions in this situation should fail with message that excluded filed not exist.

#### Example

```java
    class Book {
        String title;
    }

   Book aBook = ...

   assertThat(aBook).hasNoNullFieldsOrPropertiesExcept("name");
```
Assertions should fail because class `Book` not contains field `name`

**Comment by sustc11810424** [comment]
May I take this one? Perhaps (maybe not necessary) we can add an option 
```java
.hasNoNullFieldsOrPropertiesExcept(..., boolean strict)
```
which does what @slawekjaranowski suggests but with *strict* default to false in case we don't want a big change.

**Comment by slawekjaranowski** [comment]
Maybe change like this break some compatibility but show places where we have outdated (or wrong) tests and it will easy to fix.

**Comment by joel-costigliola** [comment]
I agree with failing the assertion if given an unknown field, no need for s strict boolean argumment for that.
